inventory/item/item.go

```diff
@@ -0,0 +1,81 @@
+package item
+
+import "inventory/inventory/constants"
+
+type BaseItem struct {
+	Name     string
+	Price    float64
+	Quantity int
+	Tax      float64
+}
+
+//every type of item should implement this
+type Item interface {
+	GetDetails() BaseItem
+	Calc() float64
+}
+
+//function to check negetive number
+func checkneg(cur float64) bool {
+	return cur <= 0
+}
+
+//abstract strct for concrete item
+type AbstractItem struct {
```

Comment:
why are we introducing one layer of AbstractItem instead of directly using BaseItem struct